langlib/lang.array/src/main/ballerina/array.bal

```diff
@@ -70,6 +88,33 @@ public isolated function enumerate(Type[] arr) returns [int, Type][] = @java:Met
 
 # Applies a function to each member of an array and returns an array of the results.
 #
+# ```ballerina
+# string[] greetings = ["Hello", "Bonjour", "Hola", "Ciao"];
+# int[] lengths = greetings.'map((greeting) => greeting.length());
+#
+# Employee[] employees = [
+#     {firstName: "Jo", lastName: "David", id: 2121},
+#     {firstName: "Emma", id: 2122},
+#     {firstName: "John", lastName: "Doe", id: 2123}
+# ];
+# string[] fullNames =
+#     employees.'map(isolated function (Employee employee) returns string {
+#         string? lastName = employee.lastName;
+#
+#         if lastName is () {
+#             return employee.firstName;
+#         }
+#
+#         return string `${employee.firstName} ${lastName}`;
+#     });
+#
+# type Employee record {
+#     string firstName;
```

Comment:
Imo, this should be define before the usage. it will improve user experience. wdyt?

Reply:
This was done intentionally since the idea is to be able to quickly understand how to use the function. By adding the type at the end, the user has to focus on the type only if they really want to. WDYT?

Reply:
+1